### PR TITLE
Header nav loading adjustments

### PR DESCRIPTION
**Pull request by DriftingSands**
header blog tag links are now gotten from the nav page content.
now preloading nav page.
adjusted header and CSS so it looks more natural while loading in.

Test URLs:
- Before: https://main--www--headwire-edge-delivery.hlx.page/
- After: https://header-delay-fix--www--headwire-edge-delivery.hlx.page/

### CODE REVIEW

File: blocks/formspree/formspree.js [hunk 1]
```diff
@@ -22,4 +22,5 @@ export default async function decorate(block) {
       <input type="submit" value="Submit">
     </form>
   `);
+  // block.innerHTML=''
```
Comment (icaraps) [inline blocks/formspree/formspree.js]: leftover

File: blocks/header/header.css [hunk 2]
```diff
@@ -312,6 +308,19 @@ header nav .nav-tools {
 
 
 @media (min-width: 900px) {
+  header nav .nav-brand:not(:has(img)) {
```
Comment (icaraps) [inline blocks/header/header.css]: `:has` is not supported on firefox yet https://caniuse.com/css-has

File: blocks/header/header.css [hunk 3]
```diff
@@ -421,13 +421,22 @@ header nav .nav-tools {
...
   header nav .nav-sections > ul > li > ul > li {
     padding: 8px 0;
   }
+
+  header nav .nav-sections .button-dropdown-wrapper .tag-menu > ul {
+    height: auto;
+    padding: .6em 1em;
+  }
+
+  header nav .nav-sections .button-dropdown-wrapper .tag-menu > ul > li:has(.mobile-back-button) {
```
Comment (icaraps) [inline blocks/header/header.css]: :has ditto

File: blocks/header/header.js [hunk 4]
```diff
@@ -1,9 +1,33 @@
...
+          <span class="nav-hamburger-icon"></span>
+        </button>
+      </div>
+      <div class="nav-brand" style="width: 80px; height: 99px;">
+        <p>
+        </p>
+      </div>
+      <div class="nav-sections" style="width: 424px; height: 38px;">
+        <ul>
+          <li>test</li>
```
Comment (icaraps) [inline blocks/header/header.js]: Do you really need to put content as the `<ul>` ? Is this required to minimize the layout shift ?

File: head.html [hunk 5]
```diff
@@ -1,6 +1,8 @@
-<meta name="viewport" content="width=device-width, initial-scale=1"/>
-<link rel="preload" href="/query-index.json" as="fetch" type="application/json" crossorigin="anonymous">
+<meta name="viewport" content="width=device-width, initial-scale=1" />
+<link rel="preload" href="/query-index.json" as="fetch" type="application/json" crossorigin="anonymous" />
+<link rel="preload" href="/nav.plain.html" as="fetch" type="application/json" crossorigin="anonymous" />
```
Comment (icaraps) [inline head.html]: Are you sure this works ? It's `text/html` not `application/json` ...

**Review by icaraps — CHANGES_REQUESTED** [review]

File: head.html (hunk 5 above)
Comment (DriftingSands) [reply head.html]: yes it still worked. Also works without a type at all.

File: blocks/header/header.js (hunk 4 above)
Comment (DriftingSands) [reply blocks/header/header.js]: removed the `<li>` tags, and added the style to the `<ul>`. No layout shift and no content

File: blocks/header/header.css (hunk 2 above)
Comment (DriftingSands) [reply blocks/header/header.css]: I removed all `:has` except for this one. I couldn't think of a simple way of doing this. Firefox users will just have a small layout shift before the image is loaded

**Pull request merged**